src/compiler/transformers/ts.ts

```diff
@@ -3239,14 +3214,14 @@ namespace ts {
         }
 
         function trySubstituteDecoratedClassName(node: Identifier): Expression {

```

Comment:
You may want to rename this function since it is not just decorated class name anynmore